draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -559,22 +560,23 @@ To this end, Verifiers MUST follow the following steps (or equivalent):
     2. Validate the signature over the SD-JWT.
     3. Validate the Issuer of the SD-JWT and that the signing key belongs to this Issuer.
     4. Check that the SD-JWT is valid using `nbf`, `iat`, and `exp` claims, if provided in the SD-JWT, and not selectively disclosed.
-    5. Check that the `_sd_alg` claim is present and its value is understood and the hash algorithm is deemed secure.
- 4. Create a copy of the SD-JWT payload, if required for further processing.
- 5. Process the Disclosures. For each Disclosure provided:
-    1. Calculate the digest over the base64url string as described in (#hashing_disclosures).
-    2. Find all `_sd` keys in the SD-JWT payload. For each such key:
+    5. Check that the `_sd_alg` claim value is understood and the hash algorithm is deemed secure.
+ 4. Process the Disclosures and `_sd` keys in the SD-JWT as follows:
+    1. Create a copy of the SD-JWT payload, if required for further processing.
+    2. For each Disclosure provided:
+       1. Calculate the digest over the base64url string as described in (#hashing_disclosures).
+    3. Find all `_sd` keys in the SD-JWT payload. For each such key:
        1. If the key does not refer to an array, the Verifier MUST reject the Presentation.
-       2. Otherwise, process each digest in the `_sd` array as follows:
-          1. Find the Disclosure referenced by the digests. If no such Disclosure can be found, the Verifier MUST reject the presentation.
+       2. Otherwise, process each entry in the `_sd` array as follows:
+          1. Using the previously calculated digests, find the matching Disclosure. If no such Disclosure can be found, the digest MUST be ignored.
```

Comment:
```suggestion
          1. Compare the value with the digest calculated in an earlier processing step, and find the matching Disclosure. If no such Disclosure can be found, the digest MUST be ignored.
```

Reply:
I think digest would be plural in that suggested change? i.e for the value from the _sd array, compare it to each of the calculated Disclosure digests to find the match. 

"Compare the value with the digests calculated in an earlier processing step..."

Reply:
```suggestion
          1. Compare the value with the digests calculated in an earlier processing step, and find the matching Disclosure. If no such Disclosure can be found, the digest MUST be ignored.
```

Reply:
In *an* earler processing step is a bit unclear. How about just "previously"?